packages/core/src/pubsub/__tests__/pubsub-keepalive.test.ts

```diff
@@ -30,16 +31,22 @@ test('publish keepalive', async () => {
     pubsubLogger,
     diagnosticsLogger
   )
-  const pubsubWithKeepalive = new PubsubKeepalive(pubsub, keepaliveInterval)
+  const maxIntervalWithoutKeepalive = 50
+  const pubsubWithKeepalive = new PubsubKeepalive(
+    pubsub,
+    keepaliveInterval,
+    maxIntervalWithoutKeepalive
+  )
   const subscription = pubsubWithKeepalive.subscribe()
 
   await delay(keepaliveInterval * 5)
 
-  expect(ipfs.pubsub.publish.mock.calls.length).toBeGreaterThan(4)
+  expect(ipfs.pubsub.publish.mock.calls.length).toBeGreaterThanOrEqual(8)
```

Comment:
this test isn't really testing the nuances of the logic here. It's also conflating testing the max time without any message with testing the max time without a keepalive.  This should be broken up into a bunch of smaller tests testing specific pieces of the logic.  i.e. We've had *no* messages over a period of time, we've had some messages but not keepalive messages, we've had keepalive messages, etc.

Reply:
hmm makes sense, will do